modules/http/src/main/scala/ch/epfl/bluebrain/nexus/service/http/routes/StaticResourceRoutes.scala

```diff
@@ -1,70 +1,57 @@
 package ch.epfl.bluebrain.nexus.service.http.routes
 
-import java.io.File
 import java.util.regex.Pattern.quote
 
 import akka.http.scaladsl.model.{StatusCodes, Uri}
 import akka.http.scaladsl.server.Directives._
 import akka.http.scaladsl.server.Route
 import ch.epfl.bluebrain.nexus.commons.http.JsonLdCirceSupport._
+import ch.epfl.bluebrain.nexus.service.http.directives.PrefixDirectives
 import io.circe.Json
 import io.circe.parser.parse
 
 import scala.io.Source
 
 /**
   * Routes that expose static resources provided on the classpath
-  * @param resourcePath path to the folder in classpath to expose
+  * @param resourcePaths [[Map]] containing mapping between path at which resource will be available(starting with `/`)
+  *                     and the path where the resource can be found on the classpath.
   * @param prefix       prefix to prepend to the routes
   * @param baseUri      base URI to use in IDs, will replace {{base}} in all the resources
   */
-class StaticResourceRoutes(resourcePath: String, prefix: String, baseUri: Uri) {
+class StaticResourceRoutes(resourcePaths: Map[String, String], prefix: String, baseUri: Uri) extends PrefixDirectives {
 
-  private def contentOf(file: File): String =
-    Source.fromFile(file).mkString
+  private def contentOf(file: String): String =
+    Source.fromInputStream(getClass.getResourceAsStream(file)).mkString
```

Comment:
I believe `Source.fromInputStream` doesn't close the Inputstream.

https://groups.google.com/forum/#!topic/scala-user/nB3kDjyoYDQ